components/teams-service/storage/postgres/postgres.go

```diff
@@ -362,24 +365,62 @@ func (p *postgres) PurgeUserMembership(ctx context.Context, userID string) ([]st
 // from a different context we should apply project filtering in this function as well.
 // Not doing it to save us a database call since it's not needed currently.
 func (p *postgres) AddUsers(ctx context.Context,
-	id string, userIDs []string) (storage.Team, error) {
-	var t storage.Team
-	err := p.db.QueryRowContext(ctx,
-		`WITH moved_row_ids AS (
-			INSERT INTO teams_users_associations (team_db_id, user_id, created_at)
+	id string, userIDs []string) (updatedUserIDs []string, err error) {
+
+	tx, err := p.db.BeginTx(ctx, nil)
+	if err != nil {
+		return nil, p.processError(err)
+	}
+
+	projectsFilter, err := ProjectsListFromContext(ctx)
+	if err != nil {
+		return nil, p.processError(err)
+	}
+
+	var dbID int
+	var teamProjects []string
+	// ensure the team exists and isn't filtered out by the project filter
+	err = p.db.QueryRowContext(ctx,
+		`SELECT db_id, projects FROM teams
```

Comment:
This function now completes the following queries in this order:
within transaction:
- SELECT the team with the projects filter to verify that it exists within the filter
- INSERT INTO user IDs in the team_user_association table, doing nothing on conflict
- UPDATE the team's `updated_at` field

after that transaction completes, we fetch the user IDs from the team.

Open to feedback on making this more concise!

Reply:
Same transaction comment for this function.

Reply:
Right now I've got the first three queries in a transaction, but I had to move the SELECTing of the final user IDs outside of the transaction since it wasn't picking up the changes to the user table.

Reply:
I exercised the raw SQL inside a transaction in an interactive session and it definitely picks up the changes if the final query is inside the transaction (as it is supposed to). When run from Go code, as you observed, it is not. That tells us that the Go interface is the culprit, and indeed it is: you were not asking for data from the transaction.
Change the line above from this:
```
p.db.QueryRowContext(ctx,
```
to this:
```
tx.QueryRowContext(ctx,
```